RIOT-OS/RIOT
# Add new board Seeeduino Xiao

### Contribution description



Board files for the Seeeduino Xiao (https://www.seeedstudio.com/Seeeduino-XIAO-Arduino-Microcontroller-SAMD21-Cortex-M0+-p-4426.html)

Basically directly copied the files from the SAMD21 Xplained Pro board and modified the GPIO and peripherals to match the different board and chip.


### Testing procedure



Couldn't find a test procedure for adding new boards, but I would be happy to run any tests you want me to. I ran `make static-test` and everything checked out (except for the doxygen check, but those are files I didn't touch)


### Issues/PRs references
